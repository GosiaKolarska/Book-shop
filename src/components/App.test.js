import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App tests', () => {

  // it('App renders without a problem', () => {

  //symulujemy element DOM
  //   const div = document.createElement('div');
  //   ReactDOM.render(<App />, div);

  //by ograniczycv zucycie pamieci przez testy, musimy odpiac komponent app od tego utoworzonego diva, element zostanie zniszczony 
  //   ReactDOM.unmountComponentAtNode(div);
  // })

  it('Snapshot matches', () => {

    //przekazujemy komponent ktory chcemy wyrenderowac (App)
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })

  it('Child components render', () => {
    const wrapper = shallow(<App />);
    // console.log(wrapper.debug());
    expect(wrapper.find('Header').exists()).toBe(true);
    expect(wrapper.find('Order').exists()).toBe(true);
    expect(wrapper.find('AdminPanel').exists()).toBe(true);
    expect(wrapper.find('Inventory').exists()).toBe(true);
  })

})
