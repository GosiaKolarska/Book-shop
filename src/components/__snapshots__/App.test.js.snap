// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <div
          className="row"
>
          <Order
                    order={Array []}
                    removeFromOrder={[Function]}
          />
          <Inventory
                    addToOrder={[Function]}
                    books={undefined}
          />
</div>,
      ],
      "className": "app container-fluid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Order
              order={Array []}
              removeFromOrder={[Function]}
/>,
            <Inventory
              addToOrder={[Function]}
              books={undefined}
/>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "order": Array [],
              "removeFromOrder": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addToOrder": [Function],
              "books": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <div
            className="row"
>
            <Order
                        order={Array []}
                        removeFromOrder={[Function]}
            />
            <Inventory
                        addToOrder={[Function]}
                        books={undefined}
            />
</div>,
        ],
        "className": "app container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Order
                order={Array []}
                removeFromOrder={[Function]}
/>,
              <Inventory
                addToOrder={[Function]}
                books={undefined}
/>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "order": Array [],
                "removeFromOrder": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addToOrder": [Function],
                "books": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
